### PR TITLE
EVEREST-239 Added cluster-info endpoint

### PR DESCRIPTION
**Pull request by gen1us2k**

### CODE REVIEW

**Review by oksana-grishchenko — APPROVED** [review]
LGTM, however with the Everest provisioned from the latest `cli` `main` the API returns
```
{
    "message": "storageclasses.storage.k8s.io is forbidden: User \"system:serviceaccount:percona-everest:everest-admin\" cannot list resource \"storageclasses\" in API group \"storage.k8s.io\" at the cluster scope"
}
```
although the rights seems to be given🤔 Just wanted to let you know.

**Comment by gen1us2k** [comment]
@oksana-grishchenko https://github.com/percona/percona-everest-cli/blob/main/pkg/install/operators.go#L877 please update your CLI =)

File: api/cluster-info.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+// GetKubernetesClusterInfo returns the cluster type and storage classes of a kubernetes cluster.
+func (e *EverestServer) GetKubernetesClusterInfo(ctx echo.Context, kubernetesID string) error {
+	_, kubeClient, code, err := e.initKubeClient(ctx, kubernetesID)
+	if err != nil {
+		return ctx.JSON(code, Error{Message: pointer.ToString(err.Error())})
+	}
+	clusterType, err := kubeClient.GetClusterType(ctx.Request().Context())
+	if err != nil {
+		return ctx.JSON(code, Error{Message: pointer.ToString(err.Error())})
```
Comment (michal-kralik) [inline api/cluster-info.go:18]: Let's please return a meaningful message to the use and log the error.

File: api/cluster-info.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	}
+	storagesList, err := kubeClient.GetStorageClasses(ctx.Request().Context())
+	if err != nil {
+		return ctx.JSON(code, Error{Message: pointer.ToString(err.Error())})
```
Comment (michal-kralik) [inline api/cluster-info.go:22]: Let's please return a meaningful message to the use and log the error.

**Review by michal-kralik — CHANGES_REQUESTED** [review]

File: api/cluster-info.go (hunk 1 above)
Comment (gen1us2k) [reply api/cluster-info.go]: Yeah. Makes sense. Thanks

File: api/cluster-info.go (hunk 2 above)
Comment (gen1us2k) [reply api/cluster-info.go]: Yeah. Makes sense. Thanks

File: api/cluster-info.go [hunk 3]
```diff
@@ -11,15 +11,18 @@ import (
 func (e *EverestServer) GetKubernetesClusterInfo(ctx echo.Context, kubernetesID string) error {
 	_, kubeClient, code, err := e.initKubeClient(ctx, kubernetesID)
 	if err != nil {
-		return ctx.JSON(code, Error{Message: pointer.ToString(err.Error())})
+		e.l.Error(err)
+		return ctx.JSON(code, Error{Message: pointer.ToString("failed building connection to the Kubernetes cluster")})
```
Comment (michal-kralik) [inline api/cluster-info.go:15]: Let's keep consistent styles by capitalizing the first letter.

File: api/cluster-info.go [hunk 4]
```diff
@@ -11,15 +11,18 @@ import (
...
 	}
 	clusterType, err := kubeClient.GetClusterType(ctx.Request().Context())
 	if err != nil {
-		return ctx.JSON(code, Error{Message: pointer.ToString(err.Error())})
+		e.l.Error(err)
+		return ctx.JSON(code, Error{Message: pointer.ToString("failed getting Kubernetes cluster provider")})
```
Comment (michal-kralik) [inline api/cluster-info.go:20]: The http status code is not correct here. Let's use `500`

File: api/cluster-info.go [hunk 5]
```diff
@@ -11,15 +11,18 @@ import (
...
 	}
 	storagesList, err := kubeClient.GetStorageClasses(ctx.Request().Context())
 	if err != nil {
-		return ctx.JSON(code, Error{Message: pointer.ToString(err.Error())})
+		e.l.Error(err)
+		return ctx.JSON(code, Error{Message: pointer.ToString("failed getting storage classes")})
```
Comment (michal-kralik) [inline api/cluster-info.go:25]: The http status code is not correct here. Let's use `500`

**Review by michal-kralik — CHANGES_REQUESTED** [review]

**Review by michal-kralik — APPROVED** [review]

**Pull request merged**